csrc/scheduler/matmul_utils.cpp

```diff
@@ -307,32 +309,63 @@ std::string getMatmulCompileTimeRejectReason(Fusion* fusion) {
   // 1. check if there is exactly one MmaOp defined in the fusion
   // 2. check if MmaOp inputs match any of supported inputs layout
   // 3. check if fusion represents expressions that are recognized by matmul
-  //    scheduler
+  // scheduler In addition to the above, it may be the case that there is no mma
+  // op, but a mul-sum pair which may be replaced by a mma op. In that case we
+  // run a pass through the IR to find such a pair of mul-sum, and the
+  // properties of that pair. We then test if the mma op that would replace the
+  // mul-sum pair can be scheduled.
 
   // #1
   auto mma_exprs = ir_utils::getOpsOfType<MmaOp>(fusion);
+  // Initializing the machinery to check if there's a Mul-Sum pair
+  // can be replaced by a Mma Op.
+  mma_utils::CombineMulSum combiner(fusion);
+  std::vector<mma_utils::MmaOrMulSumProperties> mma_from_mul_sums = {};
+  mma_utils::MmaOrMulSumProperties::InputsOutputs mma_inputs_out;
+
   if (mma_exprs.size() != 1) {
-    std::stringstream ss;
-    ss << "Matmul scheduler supports fusions only with a single MMA op, got: "
-       << mma_exprs.size();
-    return ss.str();
+    // If there are no mma ops, check if there's a suitable mul-sum pair
+    // that can be substituted by a mma op.
+    if (mma_exprs.empty()) {
+      mma_from_mul_sums = combiner.generateMulSumCanidates();
+    }
+    // There are more than 1 mma op, or there no mma ops and no suitable mul-sum
+    // pair.
+    if (mma_from_mul_sums.size() != 1) {
+      std::stringstream ss;
+      ss << "Matmul scheduler supports fusions only with a single MMA op, got: "
+         << mma_exprs.size();
+      return ss.str();
+    }
+  }
+
+  if (!mma_exprs.empty()) {
+    mma_inputs_out = {
+        static_cast<TensorView*>(mma_exprs.front()->inA()),
+        static_cast<TensorView*>(mma_exprs.front()->inB()),
+        static_cast<TensorView*>(mma_exprs.front()->out())};
```

Comment:
nit: 
```suggestion
  mma_utils::MmaOrMulSumProperties::InputsOutputs mma_inputs_out =
    mma_exprs.empty() ? mma_from_mul_sums.front().insouts :
      {
        static_cast<TensorView*>(mma_exprs.front()->inA()),
        static_cast<TensorView*>(mma_exprs.front()->inB()),
        static_cast<TensorView*>(mma_exprs.front()->out())};
```

And replace all the `mma_exprs.empty() ? ...` below with just `mma_inputs_out`